extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/binder/mpmetrics/MicroprofileMetricsProcessor.java

```diff
@@ -62,6 +62,25 @@ AdditionalBeanBuildItem registerBeanClasses() {
                 .build();
     }
 
+    @BuildStep(onlyIf = MicroprofileMetricsEnabled.class)
+    void logWarningForMpMetricsUsage(CombinedIndexBuildItem combinedIndexBuildItem) {
+        IndexView index = combinedIndexBuildItem.getIndex();
+        boolean mpMetricsPresent = false;
+
+        // Find usage of MP Metrics annotations
+        for (DotName annotation : MetricDotNames.individualMetrics) {
+            if (index.getAnnotations(annotation).size() > 0) {
+                mpMetricsPresent = true;
+                break;
+            }
+        }
+
+        if (mpMetricsPresent) {
+            log.warn("Use of MicroProfile Metrics annotations with the `quarkus-micrometer` extension in future Quarkus " +
+                    "versions will require using the `quarkus-smallrye-metrics` extension instead.");
+        }
```

Comment:
We can't word the warning this way unless/until the smallrye-metrics and micrometer extensions _can_ be enabled at the same time.  What might be better is to warn that the MP Metrics API is being supported in a compatibility mode, which may be subject to change over time. (or something).